core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -242,6 +248,16 @@
           (log/error ex "Uncaught exception processing message")
           {::severity :error, ::error-code "XX000"})))))
 
+(defn validate-and-refresh-token [{:keys [conn-state ^Authenticator authn]}]
+  (when-let [^OAuthResult authn-state (:authn-state @conn-state)]
+    (some->> (.revalidate authn authn-state) (swap! conn-state assoc :authn-state))))
```

Comment:
not a fan of using threading macros when `swap!`'s the target, the reader has to reason about what's happening - try to limit threading macros to cases when you're transforming a value.

`when-let` is likely much clearer here